### ISSUE TITLE
SMP failure for GHC 8.4

### ISSUE THREAD

**Issue by deepfire** [sol/hpack#276]
```
[ 4 of 40] Compiling Data.Aeson.Config.Types ( src/Data/Aeson/Config/Types.hs, dist/build/spec/spec-tmp/Data/Aeson/Config/Types.dyn_o )

src/Data/Aeson/Config/Types.hs:15:55: error:
    • No instance for (Semigroup (List a))
        arising from the 'deriving' clause of a data type declaration
      Possible fix:
        use a standalone 'deriving instance' declaration,
          so you can specify the instance context yourself
    • When deriving the instance for (Monoid (List a))
   |
15 |   deriving (Eq, Show, Functor, Foldable, Traversable, Monoid)
   |                                                       ^^^^^^
```
..and..
```
[17 of 40] Compiling Hpack.Syntax.Dependency ( src/Hpack/Syntax/Dependency.hs, dist/build/spec/spec-tmp/Hpack/Syntax/Dependency.dyn_o )

src/Hpack/Syntax/Dependency.hs:35:23: error:
    • No instance for (Semigroup Dependencies)
        arising from the 'deriving' clause of a data type declaration
      Possible fix:
        use a standalone 'deriving instance' declaration,
          so you can specify the instance context yourself
    • When deriving the instance for (Monoid Dependencies)
   |
35 | } deriving (Eq, Show, Monoid)
   |                       ^^^^^^
```

**Comment by DanBurton** [comment]
bump. ghc 8.4 is now released. Build failure as seen on the stackage build server:

```
[ 4 of 20] Compiling Data.Aeson.Config.Types ( src/Data/Aeson/Config/Types.hs, dist/build/Data/Aeson/Config/Types.o )

src/Data/Aeson/Config/Types.hs:15:55: error:
    • No instance for (Semigroup (List a))
        arising from the 'deriving' clause of a data type declaration
      Possible fix:
        use a standalone 'deriving instance' declaration,
          so you can specify the instance context yourself
    • When deriving the instance for (Monoid (List a))
   |
15 |   deriving (Eq, Show, Functor, Foldable, Traversable, Monoid)
   |                                                       ^^^^^^
```

Add semigroup instances.
See also: https://ghc.haskell.org/trac/ghc/wiki/Migration/8.4#SemigroupMonoidsuperclasses

**Comment by sol** [comment]
Fix released to Hackage as `0.28.1`.
